backend/courses/util.py

```diff
@@ -459,3 +459,17 @@ def subquery_count_distinct(subquery, column):
         ),
         0,
     )
+
+
+def does_object_pass_filter(obj, filter):
+    """
+    Returns True iff the given obj satisfies the given filter dictionary.
```

Comment:
Would it be helpful to add a note specifying that this really only works when the filter does not involve something like `gt`? If you included a filter like `user__id__gt` I believe this function would try to `getattr(actual_value, 'gt')` which would fail.

Reply:
Yeah very good point. Updated the docstring to reflect this (and also the fact that it can only traverse a relation to a single related object, not a related set).